packages/tracing/src/browser/browsertracing.ts

```diff
@@ -61,12 +61,17 @@ export interface BrowserTracingOptions extends RequestInstrumentationOptions {
   markBackgroundTransactions: boolean;
 
   /**
-   * beforeNavigate is called before a pageload/navigation transaction is created and allows for users
-   * to set custom transaction context. Default behavior is to return `window.location.pathname`.
+   * beforeNavigate is called before a pageload/navigation transaction is created and allows users to modify transaction
+   * context data, or drop the transaction entirely (by setting `sampled = false` in the context).
    *
-   * If undefined is returned, a pageload/navigation transaction will not be created.
+   * Note: For legacy reasons, transactions can also be dropped by returning `undefined`, though that option may
+   * disappear in the future.
+   *
+   * @param context: The context data which will be passed to `startTransaction` by default
+   *
+   * @returns A (potentially) modified context object, with `sampled = false` if the transaction should be dropped.
    */
-  beforeNavigate(context: TransactionContext): TransactionContext | undefined;
+  beforeNavigate?(context: TransactionContext): TransactionContext | undefined;
```

Comment:
This looks fine and all, but why are we now making this optional replacing the previous implementation that used a default value?

Is it related to this below note in the PR description?

> beforeNavigate was failing to create a transaction if undefined was returned, rather than creating a transaction with sampled = false to pass the negative sampling decision to its child spans/transactions